pkg/build/generator/generator.go

```diff
@@ -110,6 +110,29 @@ func (g *BuildGenerator) FetchServiceAccountSecrets(namespace, serviceAccount st
 	return result, nil
 }
 
+// findImageChangeTrigger finds an image change trigger that has a from that matches the passed in ref
+// if no match is found but there is an image change trigger with a null from, that trigger is returned
+func findImageChangeTrigger(bc *buildapi.BuildConfig, ref *kapi.ObjectReference) *buildapi.ImageChangeTrigger {
+	var defaultImageChange *buildapi.ImageChangeTrigger
+	if ref == nil {
+		return nil
+	}
+	for _, trigger := range bc.Spec.Triggers {
+		if trigger.Type != buildapi.ImageChangeBuildTriggerType {
+			continue
+		}
+		imageChange := trigger.ImageChange
+		if imageChange.From == nil {
+			defaultImageChange = imageChange
+			continue
+		}
+		if imageChange.From.Name == ref.Name && imageChange.From.Namespace == ref.Namespace {
```

Comment:
need to consider the case where the imageChange.From.Namespace is "" (implying same namespace a the BC) and ref.Namespace is set, but matches the implied namespace.

and vice versa.